extern/open.php

```diff
@@ -2,10 +2,12 @@
 require_once '../civicrm.config.php';
 require_once 'CRM/Core/Config.php';
 require_once 'CRM/Core/Error.php';
-require_once 'CRM/Utils/Array.php';
+require_once 'CRM/Utils/Type.php';
+require_once 'CRM/Utils/Rule.php';
+require_once 'CRM/Utils/Request.php';
 
 $config = CRM_Core_Config::singleton();
-$queue_id = CRM_Utils_Array::value('q', $_GET);
+$queue_id = CRM_Utils_Request::retrieveValue('q', 'positive', NULL, FALSE, 'GET');
```

Comment:
@seamuslee001 I tested this locally but it needs to be 'Positive' or it fails

Reply:
thanks @eileenmcnaughton have updated now

Reply:
Looks good to go now - I note that when you pass a string it drops it & then declares 'missing input parameters' rather than the validation error bubbling up. I don't really have an opinion on that though - at the margins it could cause a debugger to take a fee more minutes debugging